x-pack/plugins/security_solution/public/cloud_posture/pages/findings/findings_search_bar.tsx

```diff
@@ -0,0 +1,187 @@
+import { useKibana } from '../../../common/lib/kibana';
+import { CSPFinding, FetchState } from './types';
+
+// TODO: find kibanas' equivalent fn
+const isNonNullable = <T extends unknown>(v: T): v is NonNullable<T> =>
+  v !== null && v !== undefined;
+
+interface BaseFindingsSearchBarProps {
+  dataView: DataView;
+  onError(v: string): void;
+  onSuccess(v: CSPFinding[]): void;
+  onLoading(): void;
+}
+
+type FindingsSearchBarProps = BaseFindingsSearchBarProps & FetchState<CSPFinding[]>;
+
+type URLState = Parameters<NonNullable<SearchBarProps['onQuerySubmit']>>[0];
+
+const getDefaultQuery = (): Required<URLState> => ({
+  query: { language: 'kuery', query: '' },
+  dateRange: {
+    from: 'now-15m',
+    to: 'now',
+  },
+});
+
+/**
+ * Temporary Search Bar using x-pack/plugins/data
+ *
+ * TODO:
+ *  - use SiemSearchBar / QueryBar or something else ?
+ */
+export const FindingsSearchBar = ({
+  dataView,
+  loading,
+  onError,
+  onLoading,
+  onSuccess,
+}: FindingsSearchBarProps) => {
+  const { data: dataService } = useKibana().services;
+  const [filters, setFilters] = useState<Filter[]>([]);
+  const searchState = useSearchState();
+  const history = useHistory();
+
+  const {
+    ui: { SearchBar },
+    query,
+    search,
+  } = dataService;
+
+  /*
+   * This sends a query using esClient
+   * TODO:
+   * - AbortController
+   */
+  const runSearch = useCallback(async () => {
+    if (!dataView) return;
+
+    onLoading();
+
+    query.queryString.setQuery(searchState.query || getDefaultQuery().query);
+
+    const timefilter = query.timefilter.timefilter.createFilter(dataView, searchState.dateRange);
+
+    query.filterManager.setFilters([...filters, timefilter].filter(isNonNullable));
+
+    try {
+      const findingsSearchSource = await search.searchSource.create({
+        filter: query.filterManager.getFilters(),
+        query: query.queryString.getQuery(),
+        index: dataView,
+        size: 1000, // TODO: async pagination
+      });
+
+      const findingsResponse: IKibanaSearchResponse<SearchResponse<CSPFinding>> =
+        await findingsSearchSource.fetch$().toPromise();
+      onSuccess(findingsResponse.rawResponse.hits.hits.map((v) => v._source).filter(isNonNullable));
+    } catch (e) {
+      onError(e);
+    }
+  }, [
+    dataView,
+    onLoading,
+    query.queryString,
+    query.timefilter.timefilter,
+    query.filterManager,
+    searchState.query,
+    searchState.dateRange,
+    filters,
+    search.searchSource,
+    onSuccess,
+    onError,
+  ]);
+
+  /**
+   * This changes the URL which triggers a new search
+   * TODO:
+   * - make filters part of the query
+   */
+  const handleQuerySubmit = useCallback(
+    (v: URLState) => {
+      // TODO: use util fn to build query (not with URLSearchParams as it escapes 'rison')
+      const next = `source=${encode(v)}`;
+
+      if (next === history.location.search.slice(1)) {
```

Comment:
```suggestion
      const nextSearch = `source=${encode(v)}`;
      const currentSearch = history.location.search.slice(1)

      if (nextSearch === currentSearch) {
```

I think it gives it some context and makes it a bit clearer why you are checking this. `history.location.search.slice(1)` is a bit of an inside knowledge.